regression-tests/mixed-as-for-variant-20-types.cpp2

```diff
@@ -1,75 +1,37 @@
 template<int I>
 struct X { operator int() const { return I; } };
 
+template <std::size_t... Is>
+auto make_variant(std::index_sequence<Is...>){
+    return std::variant<X<Is>...>{};
+}
+
+template <typename F, template<typename...> class C, typename... Ts>
+constexpr auto for_each_type(C<Ts...>, F&& fun)
+{
+    [&]<std::size_t... Is>(std::index_sequence<Is...>){
+        if constexpr ( requires { (fun(cpp2::type_it<Is, Ts>{}), ...); } ) {
+            (fun(cpp2::type_it<Is, Ts>{}), ...);
+        }
+    }(std::index_sequence_for<Ts...>());
+}
+
 main: () -> int = {
 
-    v: std::variant<X< 0>, X< 1>, X< 2>, X< 3>, X< 4>, X< 5>, X< 6>, X< 7>, X< 8>, X< 9>, 
-                    X<10>, X<11>, X<12>, X<13>, X<14>, X<15>, X<16>, X<17>, X<18>, X<19>> = X<0>();
+    v := make_variant(std::make_index_sequence<30>());
+
+    for_each_type(v, :<T>(x : T) = {
+        v&$* = X<T::index>(); std::cout << "v as X<" << std::setw(2) << T::index << "> = " << std::setw(1) << v&$* as X<T::index> as int << std::endl;
```

Comment:
```suggestion
        v&$* = X<T::index>(); std::cout << "v as X<(T::index:2)$> = (v&$* as X<T::index> as int:1)$\n";
```
IIUC, after commit e38838e678a2f52f1c811cf7cb7f819ab3925d62.

Reply:
I will try this syntax. I was not confident in using it.

Reply:
It is not supported by apple-clang-14